### PR TITLE
Allows to mock the config during tests.

### PR DESCRIPTION
**Pull request by boehmseb**
This PR adds a safe way to modify the vara config during tests.
This is achieved by replacing the config with a copy that is tuned for testing using a context manager or decorator. This also allows to write tests that need access to a more-or-less complete tool-suite and benchbuild setup.
resolves se-passau/VaRA#649

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 24
           

Complexity increasing per file
==============================
- varats/tables/table.py  4
- varats/settings.py  1
- varats/tables/tables.py  3
- tests/test_utils.py  3
- varats/tables/__init__.py  2
- varats/tools/driver_table.py  11
- varats/tables/diff_correlation_overview_table.py  6
- tests/data/test_diff_correlation_overview_table.py  3
         

Clones added
============
- varats/tables/tables.py  1
- varats/tools/driver_plot.py  1
- varats/tools/driver_table.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 13
           

Complexity increasing per file
==============================
- tests/data/test_diff_correlation_overview_table.py  3
- tests/test_utils.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 13
           

Complexity increasing per file
==============================
- tests/test_utils.py  3
- tests/data/test_diff_correlation_overview_table.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- tests/test_utils.py  3
- tests/data/test_diff_correlation_overview_table.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- tests/utils/test_experiment_util.py  1
- tests/data/test_diff_correlation_overview_table.py  2
- tests/test_utils.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- tests/utils/test_experiment_util.py  1
- tests/test_utils.py  3
- tests/data/test_diff_correlation_overview_table.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- tests/test_utils.py  3
- tests/utils/test_experiment_util.py  1
- tests/data/test_diff_correlation_overview_table.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- tests/utils/test_experiment_util.py  1
- tests/test_utils.py  3
- tests/data/test_diff_correlation_overview_table.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- tests/data/test_diff_correlation_overview_table.py  2
- tests/utils/test_experiment_util.py  1
- tests/test_utils.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- tests/test_utils.py  3
- tests/data/test_diff_correlation_overview_table.py  2
- tests/utils/test_experiment_util.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- tests/utils/test_experiment_util.py  1
- tests/data/test_diff_correlation_overview_table.py  2
- tests/test_utils.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- tests/data/test_diff_correlation_overview_table.py  2
- tests/utils/test_experiment_util.py  1
- tests/test_utils.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 8
           

Complexity increasing per file
==============================
- tests/data/test_diff_correlation_overview_table.py  2
- tests/test_utils.py  5
- tests/utils/test_experiment_util.py  1
         

Clones added
============
- varats/projects/c_projects/qemu.py  1
- varats/projects/c_projects/vim.py  2
- tests/test_utils.py  2
- varats/experiments/commit_annotation_report.py  1
- tests/utils/test_experiment_util.py  4
- varats/projects/c_projects/gravity.py  1
- varats/projects/c_projects/opus.py  1
- varats/projects/c_projects/x264.py  1
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/libvpx.py  1
         

Clones removed
==============
+ varats/experiments/vara_full_mtfa.py  -1
+ varats/experiments/blame_report_experiment.py  -1
+ varats/projects/c_projects/glibc.py  -1
+ varats/projects/c_projects/busybox.py  -1
+ varats/projects/c_projects/openssl.py  -1
+ varats/experiments/just_compile.py  -1
+ varats/projects/c_projects/gzip.py  -1
+ varats/projects/c_projects/lz4.py  -1
+ varats/experiments/commit_report_experiment.py  -2
+ varats/experiments/phasar_env_analysis.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 8
           

Complexity increasing per file
==============================
- tests/data/test_diff_correlation_overview_table.py  2
- tests/test_utils.py  5
- tests/utils/test_experiment_util.py  1
         

Clones added
============
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/opus.py  1
- varats/experiments/commit_annotation_report.py  1
- varats/projects/c_projects/gravity.py  1
- varats/projects/c_projects/vim.py  2
- varats/projects/c_projects/libvpx.py  1
- varats/projects/c_projects/qemu.py  1
- tests/test_utils.py  2
- tests/utils/test_experiment_util.py  4
- varats/projects/c_projects/x264.py  1
         

Clones removed
==============
+ varats/projects/c_projects/glibc.py  -1
+ varats/projects/c_projects/busybox.py  -1
+ varats/experiments/commit_report_experiment.py  -2
+ varats/experiments/just_compile.py  -1
+ varats/projects/c_projects/openssl.py  -1
+ varats/experiments/blame_report_experiment.py  -1
+ varats/projects/c_projects/gzip.py  -1
+ varats/experiments/phasar_env_analysis.py  -1
+ varats/projects/c_projects/lz4.py  -1
+ varats/experiments/vara_full_mtfa.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- tests/data/test_diff_correlation_overview_table.py  2
- tests/test_utils.py  5
- tests/utils/test_experiment_util.py  1
         

Clones added
============
- varats/projects/c_projects/qemu.py  1
- varats/projects/c_projects/vim.py  2
- varats/projects/c_projects/x264.py  1
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/gravity.py  1
- varats/projects/c_projects/libvpx.py  1
- varats/experiments/commit_annotation_report.py  1
- varats/projects/c_projects/opus.py  1
- tests/test_utils.py  2
- tests/utils/test_experiment_util.py  4
         

Clones removed
==============
+ varats/experiments/commit_report_experiment.py  -2
+ varats/experiments/vara_full_mtfa.py  -1
+ varats/projects/c_projects/openssl.py  -1
+ varats/projects/c_projects/lz4.py  -1
+ varats/projects/c_projects/gzip.py  -1
+ varats/experiments/just_compile.py  -1
+ varats/projects/c_projects/busybox.py  -1
+ varats/projects/c_projects/glibc.py  -1
+ varats/experiments/blame_report_experiment.py  -1
+ varats/experiments/phasar_env_analysis.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

File: tests/TEST_INPUTS/tables/b_diff_correlation_overview.tex [hunk 1]
```diff
@@ -0,0 +1,12 @@
+\begin{tabular}{lrrrrr}
+\toprule
+{} & \multicolumn{5}{c}{xz} \\
+{} & churn\_total & diff\_ci\_total & ci\_degree\_mean & author\_mean & avg\_time\_mean \\
+\midrule
+\textbf{churn\_total   } &    1.000000 &      0.084695 &       0.297545 &    0.072751 &      0.659304 \\
+\textbf{diff\_ci\_total } &    0.084695 &      1.000000 &       0.853532 &    0.874040 &      0.448876 \\
+\textbf{ci\_degree\_mean} &    0.297545 &      0.853532 &       1.000000 &    0.764138 &      0.695736 \\
+\textbf{author\_mean   } &    0.072751 &      0.874040 &       0.764138 &    1.000000 &      0.361345 \\
+\textbf{avg\_time\_mean } &    0.659304 &      0.448876 &       0.695736 &    0.361345 &      1.000000 \\
```
Comment (vulder) [inline tests/TEST_INPUTS/tables/b_diff_correlation_overview.tex]: how save are these on the test side against rounding errors?

**Review by vulder — APPROVED** [review]
This looks really nice an easy to use.

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 6
           

Complexity increasing per file
==============================
- tests/test_utils.py  5
- tests/utils/test_experiment_util.py  1
- tests/data/test_diff_correlation_overview_table.py  2
         

Clones added
============
- varats/projects/c_projects/vim.py  2
- tests/test_utils.py  2
- tests/utils/test_experiment_util.py  4
- varats/projects/c_projects/opus.py  1
- varats/projects/c_projects/x264.py  1
- varats/projects/c_projects/libvpx.py  1
- varats/projects/c_projects/qemu.py  1
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/gravity.py  1
         

Clones removed
==============
+ varats/experiments/blame_report_experiment.py  -1
+ varats/projects/c_projects/openssl.py  -1
+ varats/projects/c_projects/lz4.py  -1
+ varats/projects/c_projects/gzip.py  -1
+ varats/projects/c_projects/busybox.py  -1
+ varats/experiments/commit_report_experiment.py  -1
+ varats/projects/c_projects/glibc.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

File: tests/TEST_INPUTS/tables/b_diff_correlation_overview.tex (hunk 1 above)
Comment (boehmseb) [reply tests/TEST_INPUTS/tables/b_diff_correlation_overview.tex]: I didn't encounter any problems yet.

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- tests/test_utils.py  5
- tests/utils/test_experiment_util.py  1
- tests/data/test_diff_correlation_overview_table.py  2
         

Clones added
============
- tests/test_utils.py  2
- varats/projects/c_projects/opus.py  1
- varats/projects/c_projects/vim.py  2
- varats/projects/c_projects/gravity.py  1
- varats/projects/c_projects/qemu.py  1
- varats/projects/c_projects/libvpx.py  1
- tests/utils/test_experiment_util.py  4
- varats/projects/c_projects/x264.py  1
- varats/projects/c_projects/xz.py  3
         

Clones removed
==============
+ varats/projects/c_projects/busybox.py  -1
+ varats/projects/c_projects/lz4.py  -1
+ varats/projects/c_projects/openssl.py  -1
+ varats/experiments/blame_report_experiment.py  -1
+ varats/experiments/commit_report_experiment.py  -1
+ varats/projects/c_projects/gzip.py  -1
+ varats/projects/c_projects/glibc.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- tests/test_utils.py  5
- tests/data/test_diff_correlation_overview_table.py  2
- tests/utils/test_experiment_util.py  1
         

Clones added
============
- tests/test_utils.py  2
- varats/projects/c_projects/gravity.py  1
- varats/projects/c_projects/x264.py  1
- varats/projects/c_projects/libvpx.py  1
- varats/projects/c_projects/opus.py  1
- tests/utils/test_experiment_util.py  4
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/vim.py  2
- varats/projects/c_projects/qemu.py  1
         

Clones removed
==============
+ varats/projects/c_projects/openssl.py  -1
+ varats/experiments/blame_report_experiment.py  -1
+ varats/projects/c_projects/glibc.py  -1
+ varats/experiments/commit_report_experiment.py  -1
+ varats/projects/c_projects/gzip.py  -1
+ varats/projects/c_projects/busybox.py  -1
+ varats/projects/c_projects/lz4.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- tests/utils/test_experiment_util.py  1
- tests/data/test_diff_correlation_overview_table.py  2
- tests/test_utils.py  5
         

Clones added
============
- varats/projects/c_projects/xz.py  3
- varats/projects/c_projects/x264.py  1
- varats/projects/c_projects/opus.py  1
- varats/projects/c_projects/gravity.py  1
- varats/projects/c_projects/vim.py  2
- tests/utils/test_experiment_util.py  4
- varats/projects/c_projects/libvpx.py  1
- varats/projects/c_projects/qemu.py  1
- tests/test_utils.py  2
         

Clones removed
==============
+ varats/projects/c_projects/openssl.py  -1
+ varats/projects/c_projects/gzip.py  -1
+ varats/experiments/blame_report_experiment.py  -1
+ varats/projects/c_projects/lz4.py  -1
+ varats/experiments/commit_report_experiment.py  -1
+ varats/projects/c_projects/glibc.py  -1
+ varats/projects/c_projects/busybox.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5581031&bid=18294614)

**Review by vulder — APPROVED** [review]

**Pull request merged**